lib/auth.ts

```diff
@@ -175,17 +175,24 @@ const getAuth = function (
 		email: string;
 		password: string;
 	}): Promise<string> {
-		const { body } = await request.send<string>({
-			method: 'POST',
-			baseUrl: apiUrl,
-			url: '/login_',
-			body: {
-				username: credentials.email,
-				password: String(credentials.password),
-			},
-			sendToken: false,
-		});
-		return body;
+		try {
+			const { body } = await request.send<string>({
+				method: 'POST',
+				baseUrl: apiUrl,
+				url: '/login_',
+				body: {
+					username: credentials.email,
+					password: String(credentials.password),
+				},
+				sendToken: false,
+			});
+			return body;
+		} catch (err) {
+			if (err.statusCode === 429) {
+				throw new errors.BalenaTooManyRequests();
+			}
+			throw new errors.BalenaInvalidLoginCredentials();
```

Comment:
I think that this should be checking for a 401, to not silence different errros.

Reply:
Can we also get a test case for these plz?